geostore/populate_catalog/task.py

```diff
@@ -95,8 +95,4 @@ def handle_message(metadata_key: str) -> None:
     if root_catalog.get_child(dataset_metadata.id) is None:
         root_catalog.add_child(child=dataset_metadata, strategy=GeostoreSTACLayoutStrategy())
 
-    root_catalog.normalize_hrefs(
-        f"{S3_URL_PREFIX}{Resource.STORAGE_BUCKET_NAME.resource_name}",
-        strategy=GeostoreSTACLayoutStrategy(),
-    )
```

Comment:
Isn't there some missing code to actually replace the root `href`?

```
>               assert item_json[STAC_LINKS_KEY] == expected_item_links
E               AssertionError: assert [{'href': './CCRrWAn9FPPOyNFd1Q6I.json',\n  'rel': 'root',\n  'type': 'application/json'},\n {'href': './CCRrWAn9FPPOyNFd1Q6I.json',\n  'rel': 'parent',\n  'type': 'application/json'}] == [{'href': '../catalog.json', 'rel': 'root', 'type': 'application/json'},\n {'href': './CCRrWAn9FPPOyNFd1Q6I.json',\n  'rel': 'parent',\n  'type': 'application/json'}]
E                 At index 0 diff: {'rel': 'root', 'href': './CCRrWAn9FPPOyNFd1Q6I.json', 'type': 'application/json'} != {'rel': 'root', 'href': '../catalog.json', 'type': 'application/json'}
E                 Full diff:
E                   [
E                 -  {'href': '../catalog.json',
E                 +  {'href': './CCRrWAn9FPPOyNFd1Q6I.json',
E                     'rel': 'root',
E                     'type': 'application/json'},
E                    {'href': './CCRrWAn9FPPOyNFd1Q6I.json',
E                     'rel': 'parent',
E                     'type': 'application/json'},
E                   ]
```

Reply:
Took a while for me to trace this down. There are two options to tackle this bug: 
`root_catalog.save_object()` or `root_catalog.save(catalog_type=CatalogType.SELF_CONTAINED)` see `populate_catalog/task.py`

The first option `save_object` is the workaround suggested by Pystac (see https://github.com/stac-utils/pystac/issues/284). This would be the ideal implementation as none of the metadata gets duplicated during dataset version creation (at least in in my own tests); however, from what I can tell, this options eludes `importer` from being called in `import_metadata_file/task.py` (which is why you are seeing the error message above). Why this may be the case exactly, I am still slightly unclear. I have tried tracing this down but with no success thus far. Perhaps you might have a better understanding of how Geostore utilizes Pystac to shine some light on why this may be? I will continue to explore and will update if I am able to work this out.

The second option `root_catalog.save(catalog_type=CatalogType.SELF_CONTAINED)` is the option I went with in this PR. This appears to only be a partial fix. It prevents `item.json` from being updated with every new dataset version creation; however, it doesn't prevent `collection.json` from being updated every time a new dataset is added or updated.